### ISSUE TITLE
Replace dict.has_key() with the 'in' keyword, for twisted/words

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8360]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8360|
|Type|enhancement|
|Created|2016-05-26 17:26:13Z|
|Branch|https://github.com/twisted/twisted/tree/8360-rodrigc-has_key2|

In PEP 0290, has_key() was deprecated in Python 2.2 and higher: 
* https://www.python.org/dev/peps/pep-0290/#testing-dictionary-membership
* https://docs.python.org/2.2/whatsnew/node4.html

In Python 3, dict.has_key() was removed:
* https://docs.python.org/3.0/whatsnew/3.0.html#builtins

PEP 0290 says that for testing dictionary membership, this:

```
if d.has_key(k):
```

should be changed to this:

```
if k in d:
```

<details><summary>Searchable metadata</summary>

```
trac-id__8360 8360
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8360_rodrigc_has_key2 8360-rodrigc-has_key2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__None None
time__1464283573591397 1464283573591397
changetime__1465420919332036 1465420919332036
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/76

This was pulled out from [ticket:4053]

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @rodrigc|
|-|-|

It looks, thanks!

I have also created the topfile.

Waiting for buildbot results and will merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 30bfef26a56c3169e077b3ed176e1bd094f23846
```
#!CommitTicketReference repository="" revision="30bfef26a56c3169e077b3ed176e1bd094f23846"
Merge 8360-rodrigc-has_key2: Replace dict.has_key() in twisted.words.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8360
```
